probe-rs/src/flashing/download.rs

```diff
@@ -40,39 +46,67 @@ pub enum FileDownloadError {
     Object(&'static str),
 }
 
-/// Downloads a file at `path` into flash.
+/// Downloads a file of given `format` at `path` to the flash with progress reporting.
+///
+/// This will ensure that memory bounderies are honored and does unlocking, erasing and programming of the flash for you.
+///
+/// If `keep_unwritten_bytes` is `true`, erased portions that are not overwritten by the ELF data
+/// are restored afterwards, such that the old contents are untouched.
+///
+/// If no progress reporting is desired, have a look at `flashing::download_file()`.
 pub fn download_file_with_progress_reporting(
     session: &Session,
     path: &Path,
     format: Format,
     memory_map: &[MemoryRegion],
+    keep_unwritten_bytes: bool,
```

Comment:
I don't like boolean parameters. Maybe we could add a flash options struct, which would contain both the `keep_unwritten_bytes` setting, as well as the progress callback? Then we could also remove one of the functions.

Reply:
I changed this accordingly. There is still two fuctions, one withg options and one without, but way less arguments :)